katana/native/cases/views.py

```diff
@@ -0,0 +1,370 @@
+"""
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+"""
+
+from django.shortcuts import render, redirect
+from django.views.decorators.csrf import csrf_protect, csrf_exempt
+from django.template.context_processors import csrf
+import os, sys, glob, copy 
+from collections import OrderedDict
+from django.http import HttpResponse, JsonResponse
+from django.template import loader, RequestContext
+from xml.sax.saxutils import escape, unescape
+import xml.dom.minidom 
+import xmltodict
+from django.core.serializers import serialize
+from django.db.models.query import QuerySet
+from django.template import Library
+import json
+from utils.navigator_util import Navigator
+from katana_utils import *
+import scanfiles
+
+
+navigator = Navigator();
+path_to_src_python_file = navigator.get_katana_dir() + os.sep + "config.json"
+All_case_action_details = py_file_details(json.loads(open(path_to_src_python_file).read())['pythonsrcdir']);
+setPythonSrcDir(All_case_action_details);
+
+def index(request):
+	path_to_config_file = navigator.get_katana_dir() + os.sep + "config.json"
+	x= json.loads(open(path_to_config_file).read());
+	path_to_testcases = x['xmldir'];
+	template = loader.get_template("listAllCases.html")
+	fpath = path_to_testcases ;
+
+	jtree = navigator.get_dir_tree_json(fpath)
+	jtree['state']= { 'opened': True };
+
+	context = { 
+		'title' 	: 'List of Cases',	
+		'dirpath' : path_to_testcases,
+		'treejs': jtree,
+	}
+	return HttpResponse(template.render(context, request))
+
+def getCaseListTree(request):
+	path_to_config_file = navigator.get_katana_dir() + os.sep + "config.json"
```

Comment:
Make it a a generic util - so that it can re-used.

Reply:
Done for cases. Repeat for projects and suites.